vnet/lib/vnet/node_api/mac_lease.rb

```diff
@@ -6,6 +6,17 @@ class MacLease < EventBase
 
     class << self
 
+      def dispatch_deleted_for_segment(segment_id, deleted_at)
+        filter_date = ['deleted_at >= ? || deleted_at = NULL',
+                       (deleted_at || Time.now) - 3]
+
+        mac_address_ids = M::MacAddress.with_deleted.where(segment_id: segment_id).map { |ma| ma.id }
+
+        M::MacLease.with_deleted.where(mac_address_id: mac_address_ids).where(*filter_date).each { |lease|
```

Comment:
I thought the ds was better because it handles everthing inside an sql query. I tried with a ds here first too but it didn't work because how of the mac_address plugin is written. I forgot what the exact error was but I can reproduce it and post it if you want to have a look.